src/inference/hmc.jl

```diff
@@ -262,6 +466,41 @@ function step(model, spl::Sampler{<:Hamiltonian}, vi::VarInfo, is_first::Val{fal
     return vi, is_accept
 end
 
+
+# Function for multiple steps
+function steps!(model, spl::Sampler{<:AdaptiveHamiltonian}, vi, samples; rng::AbstractRNG=GLOBAL_RNG)
+    ahmc_samples =  AHMC.sample(rng, spl.info[:h], spl.info[:traj], Vector{Float64}(vi[spl]), spl.alg.n_iters, spl.info[:adaptor], spl.alg.n_adapts)
+    for i = 1:length(samples)
+        vi[spl] = ahmc_samples[i]
+        samples[i].value = Sample(vi, spl).value
+    end
+end
+
+function steps!(model, spl::Sampler{<:HMC}, vi, samples; rng::AbstractRNG=GLOBAL_RNG)
+    ahmc_samples =  AHMC.sample(rng, spl.info[:h], spl.info[:traj], Vector{Float64}(vi[spl]), spl.alg.n_iters)
+    for i = 1:length(samples)
+        vi[spl] = ahmc_samples[i]
+        samples[i].value = Sample(vi, spl).value
+    end
+end
+
+function steps!(model, spl::Sampler{<:Hamiltonian}, vi, samples; rng::AbstractRNG=GLOBAL_RNG)
+    for i = 1:length(samples)
+        time_elapsed = @elapsed vi, is_accept = step(model, spl, vi, Val(i == 1))
```

Comment:
Unrolling the first iteration and using `Val(true)` and `Val(false)` will make this loop faster.

Reply:
I see. Nice point. I will change!